packages/e2e-test-utils-playwright/src/request-utils/rest.ts

```diff
@@ -22,18 +23,31 @@ function splitRequestsToChunks( requests: BatchRequest[], chunkSize: number ) {
 }
 
 async function getAPIRootURL( request: APIRequestContext ) {
-	// Discover the API root url using link header.
-	// See https://developer.wordpress.org/rest-api/using-the-rest-api/discovery/#link-header
-	const response = await request.head( WP_BASE_URL );
-	const links = response.headers().link;
-	const restLink = links?.match( /<([^>]+)>; rel="https:\/\/api\.w\.org\/"/ );
-
-	if ( ! restLink ) {
-		throw new Error( `Failed to discover REST API endpoint.
- Link header: ${ links }` );
-	}
-
-	const [ , rootURL ] = restLink;
+	let restLink: unknown = null;
+
+	// Retry until the REST API root URL is discovered.
+	// See https://github.com/WordPress/gutenberg/issues/61627
+	await expect
+		.poll(
+			async () => {
+				// Discover the API root url using link header.
+				// See https://developer.wordpress.org/rest-api/using-the-rest-api/discovery/#link-header
+				const response = await request.head( WP_BASE_URL );
+				const links = response.headers().link;
+				restLink = links?.match(
+					/<([^>]+)>; rel="https:\/\/api\.w\.org\/"/
+				);
+
+				return restLink;
+			},
+			{
+				message: 'Failed to discover REST API endpoint.',
+				timeout: 60_000, // 1 minute.
+			}
+		)
+		.not.toBeFalsy();
```

Comment:
Nit: I think `.not.toBeNull()` will be enough here.

```suggestion
		.not.toBeNull();
```